### ISSUE TITLE
Cannot obtain the same results as the paper

### ISSUE THREAD

**Issue by xiaominli1020** [rinongal/textual_inversion#152]
Training set:
![image](https://github.com/rinongal/textual_inversion/assets/25433111/b3cb1397-7696-4c8e-aa64-04c775e7d611)
Inference prompt: "A * backpack"
we get the following results:
![image](https://github.com/rinongal/textual_inversion/assets/25433111/debc4c05-7de2-410f-9188-bda40dced42d)
How to solve this issue?

**Comment by rinongal** [comment]
Hi @XiaominLi1997 ,

Just to make sure: 
1) Are you using the LDM or the Stable Diffusion version? Our results are all using LDM (the paper predates Stable Diffusion). 
2) Did you change the batch size or any other parameter? We used the original LDM code which has LR scaled by batch size, so you might get significant differences from that.
3) There's some rather significant seed sensitivity, so you might get very different results with just seed changes.

We re-ran things for a related issue here and got [this](https://github.com/rinongal/textual_inversion/files/9647761/cat_5k_26_09.zip) cat checkpoint that you could try playing with.

**Comment by xiaominli1020** [comment]
> Hi @XiaominLi1997 ,
> 
> Just to make sure:
> 
> 1. Are you using the LDM or the Stable Diffusion version? Our results are all using LDM (the paper predates Stable Diffusion).
> 2. Did you change the batch size or any other parameter? We used the original LDM code which has LR scaled by batch size, so you might get significant differences from that.
> 3. There's some rather significant seed sensitivity, so you might get very different results with just seed changes.
> 
> We re-ran things for a related issue here and got [this](https://github.com/rinongal/textual_inversion/files/9647761/cat_5k_26_09.zip) cat checkpoint that you could try playing with.

Thanks for your reply. I use Stable Diffusion version. I guess there are **_two_** possible reasons.

- I set batch_size =1. I am not sure if this can degrade the generating performance during gradient descent.

- When using Stable Diffusion version, I got the following warning.

_Some weights of the model checkpoint at openai/clip-vit-large-patch14 were not used when initializing CLIPTextModel: ['vision_model.encoder.layers.0.layer_norm2.weight', 'vision_model.encoder.layers.23.mlp.fc2.bias', ... ...]_
which means we did not load weights of openai/clip-vit-large-patch14 successfully.
How can I fix this error?
**[Training setting]**
python main.py  --base configs/stable-diffusion/v1-finetune.yaml -t --actual_resume projects/v1-5-pruned.ckpt -n toy --gpus 0, --data_root ./images/cat_statue --init_word toy --no-test
**[Inference setting]**
python stable_txt2img.py --n_samples 8 --embedding_path projects/textual_inversion/logs/cat_statue2023-06-04T06-46-55_cat/checkpoints/embeddings_gs-6099.pt --ckpt projects/v1-5-pruned.ckpt --prompt "a * backpack."

**Comment by rinongal** [comment]
The Stable Diffusion version doesn't work quite as well because SD's text encoder is much smaller and with lower dimensions. It means our learned 'words' have to be further away from real words, which leads to harder-to-edit results.

If you're trying to re-create paper results, you should use LDM (which the paper used) and *not* Stable Diffusion. 

If you just want a text-based personalization method as a basis for future work, then you can certainly build on the SD version or use newer papers (for example https://github.com/NeuralTextualInversion/NeTI).

On getting better results with SD:
I would give the automatic1111 or diffusers implementation of Textual Inversion a try, since they have optimizations that should let you run bigger batch sizes on less memory. If you are only able to fit in a batch size of 1, I would look at using gradient accumulation steps (you can adjust this in the config) and / or increasing the learning rate.

I think the weight think is a warning and not an error. Check if it says something like "This IS expected if you are initializing..." and if so you can ignore it.

**Comment by xiaominli1020** [comment]
Thank you! 

> If you just want a text-based personalization method as a basis for future work, then you can certainly build on the SD version or use newer papers (for example https://github.com/NeuralTextualInversion/NeTI).

I wanna obtain a personalized embedding of the properties of a subject (object). I will have a try on NeTI.
